packages/util-waiter/src/utils/sleep.ts

```diff
@@ -0,0 +1,18 @@
+import { AbortSignal } from "@aws-sdk/abort-controller";
+
+import { WaiterResult, WaiterState } from "../waiter";
+
+export const sleep = (seconds: number) => {
+  return new Promise((resolve) => setTimeout(resolve, seconds * 1000));
+};
+
+export const waiterTimeout = async (seconds: number): Promise<WaiterResult> => {
```

Comment:
This method and abortTimeout seem like they're not really useful on their own and publicly expose implementation details of waiters. Did you consider maybe moving these into poller as non-exported functions? You could, for example, export a function like `createWaiter` or something that takes the maxWaittime, abortController, waiter settings, etc, and it returns a "waiter" or whatever the abstraction is (a promise I guess), that automatically hooks up the promise race to timeout a waiter and checking the abort controller. I suspect this will make codegen simpler *and* limit the amount of public API surface area y'all have to maintain.

Reply:
Yeah I thought about just exposing the race. This gets kinda messy cause you have to deal with the generics exposed in the client waiter and pass them all through twice. Once to createWaiter and once again through to the underlying clientWaiter. The function def gets silly. 

My 2c is that this is probably too much abstraction and the code starts to be much less clean, but I can totally see what you're saying and I agree that it does reduce API surface area. @AllanZhengYP what do you think?